src/viewport_renderer.rs

```diff
@@ -176,18 +176,52 @@ impl fmt::Display for ViewportRendererAddGeometryError {
 
 impl error::Error for ViewportRendererAddGeometryError {}
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
Is `Hash` needed here?

Reply:
It is not. I think I read somewhere you want to by default implement everything you can guarantee will stay implemented, and this seems like it could since `Msaa` is a very simple enum. But because we are an application, we don't need to think too heavily about backwards compatibility and removing trait implementations...

Not really sure what to do. Adding it later if we need it should be ok as well. Should we remove `PartialEq` and `Eq` too then?

Reply:
I don't know what performance impact it has and couldn't quickly duckduck anything relevant. I'd remove it either way for clarity reasons. `Eq` is actually used so it cannot be removed.